### PR TITLE
HDFS-15077. Fix intermittent failure of TestDFSClientRetries#testLeas…

### PR DESCRIPTION
**Pull request by iwasakims**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  25m 18s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 33s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  18m 30s |  trunk passed  |
| +1 :green_heart: |  compile  |   3m 21s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 59s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  8s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  16m  3s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   2m  3s |  trunk passed  |
| +0 :ok: |  spotbugs  |   2m 47s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 54s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 16s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  12m 57s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 50s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  3s |  hadoop-hdfs-client in the patch passed.  |
| -1 :x: |  unit  |  94m 45s |  hadoop-hdfs in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 45s |  The patch does not generate ASF License warnings.  |
|  |   | 200m 50s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.server.namenode.TestNamenodeCapacityReport |
|   | hadoop.hdfs.server.datanode.TestNNHandlesCombinedBlockReport |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1797 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 9c1050140e57 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 59aac00 |
| Default Java | 1.8.0_232 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/3/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/3/testReport/ |
| Max. process+thread count | 4227 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-client hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  31m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  20m  4s |  trunk passed  |
| +1 :green_heart: |  compile  |   3m 39s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m  3s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 59s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  16m  2s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 52s |  trunk passed  |
| +0 :ok: |  spotbugs  |   3m  9s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 26s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 12s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 48s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  7s |  hadoop-hdfs-client in the patch passed.  |
| -1 :x: |  unit  | 142m 35s |  hadoop-hdfs in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 41s |  The patch does not generate ASF License warnings.  |
|  |   | 257m 15s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.TestDistributedFileSystemWithECFile |
|   | hadoop.hdfs.TestFileChecksumCompositeCrc |
|   | hadoop.hdfs.TestFileChecksum |
|   | hadoop.hdfs.TestReconstructStripedFile |
|   | hadoop.hdfs.TestErasureCodingPolicyWithSnapshotWithRandomECPolicy |
|   | hadoop.hdfs.TestEncryptionZonesWithKMS |
|   | hadoop.hdfs.server.datanode.TestNNHandlesCombinedBlockReport |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1797 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 18e39deaa52d 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 59aac00 |
| Default Java | 1.8.0_232 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/2/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/2/testReport/ |
| Max. process+thread count | 3513 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-client hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  31m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 10s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  19m 44s |  trunk passed  |
| +1 :green_heart: |  compile  |   3m 28s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 55s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  7s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  15m 26s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 53s |  trunk passed  |
| +0 :ok: |  spotbugs  |   2m 58s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m  9s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 44s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 57s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 12s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  0s |  hadoop-hdfs-client in the patch passed.  |
| -1 :x: |  unit  | 151m 30s |  hadoop-hdfs in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  The patch does not generate ASF License warnings.  |
|  |   | 265m  3s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.client.impl.TestBlockReaderLocal |
|   | hadoop.hdfs.TestDFSInputStream |
|   | hadoop.hdfs.TestDistributedFileSystemWithECFile |
|   | hadoop.hdfs.TestFileChecksum |
|   | hadoop.hdfs.qjournal.server.TestJournalNodeSync |
|   | hadoop.hdfs.TestDecommissionWithBackoffMonitor |
|   | hadoop.hdfs.TestDFSStripedOutputStreamWithRandomECPolicy |
|   | hadoop.hdfs.TestDeadNodeDetection |
|   | hadoop.hdfs.server.datanode.TestDataNodeUUID |
|   | hadoop.hdfs.TestParallelShortCircuitLegacyRead |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1797 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ddba568a4ba1 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 59aac00 |
| Default Java | 1.8.0_232 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/1/testReport/ |
| Max. process+thread count | 3441 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-client hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDFSClientRetries.java [hunk 1]
```diff
@@ -406,14 +407,14 @@ public void testLeaseRenewSocketTimeout() throws Exception
       // throws SocketTimeoutException.
       Mockito.doNothing().when(spyNN).renewLease(
           Mockito.anyString());
-      leaseRenewer = client.getLeaseRenewer();
-      leaseRenewer.setRenewalTime(100);
+      LeaseRenewer leaseRenewer2 = client.getLeaseRenewer();
```
Comment (aajisaka) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDFSClientRetries.java]: Should we set final to leaseRenewer2 as well as leaseRenewer1?

Comment (aajisaka) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDFSClientRetries.java]: I'm +1 if that is addressed.

Comment (iwasakims) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDFSClientRetries.java]: I addressed your comment. Thanks.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 19s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   1m  8s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  19m 39s |  trunk passed  |
| +1 :green_heart: |  compile  |   3m 36s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m  0s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  4s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  15m 43s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 58s |  trunk passed  |
| +0 :ok: |  spotbugs  |   2m 50s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 58s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 42s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 42s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 57s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 22s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 57s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 58s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  4s |  hadoop-hdfs-client in the patch passed.  |
| -1 :x: |  unit  | 146m 12s |  hadoop-hdfs in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 43s |  The patch does not generate ASF License warnings.  |
|  |   | 233m  6s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.server.namenode.ha.TestHASafeMode |
|   | hadoop.hdfs.TestReconstructStripedFile |
|   | hadoop.hdfs.TestErasureCodingPoliciesWithRandomECPolicy |
|   | hadoop.hdfs.TestDeadNodeDetection |
|   | hadoop.hdfs.server.blockmanagement.TestBlockTokenWithDFSStriped |
|   | hadoop.hdfs.TestBlockStoragePolicy |
|   | hadoop.hdfs.server.sps.TestExternalStoragePolicySatisfier |
|   | hadoop.hdfs.TestReplaceDatanodeFailureReplication |
|   | hadoop.hdfs.server.datanode.TestNNHandlesBlockReportPerStorage |
|   | hadoop.hdfs.TestFileCreation |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1797 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ac5da571e7ce 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a43c177 |
| Default Java | 1.8.0_232 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/4/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/4/testReport/ |
| Max. process+thread count | 4399 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-client hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1797/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**